keps/sig-node/5067-pod-generation/README.md

```diff
@@ -0,0 +1,931 @@
+  CRI or CNI may require updating that component before the kubelet.
+-->
+
+Previous versions of clients unaware of `generation` functionality would either 
+not set the pod `generation` field (having the effective value of 0) or set it to 
+some custom value, though the latter is unlikely. In either case, the API server
+will ignore whatever value of `generation` is set by the client and will manage 
+`generation` itself (setting it to 1 for newly created pods, or incrementing it 
+for pod updates). 
+
+If there is version skew between multiple apiservers, the `generation` may or may
+not be incremented. To address this, we will not feature-gate the logic in the
+apiserver that increments `generation`. That means that by the time 
+ObservedGeneration goes to beta, there will be 2 versions of apiservers updating
+the `generation` field, removing the issue. 
+
+## Production Readiness Review Questionnaire
+
+<!--
+
+Production readiness reviews are intended to ensure that features merging into
+Kubernetes are observable, scalable and supportable; can be safely operated in
+production environments, and can be disabled or rolled back in the event they
+cause increased failures in production. See more in the PRR KEP at
+https://git.k8s.io/enhancements/keps/sig-architecture/1194-prod-readiness.
+
+The production readiness review questionnaire must be completed and approved
+for the KEP to move to `implementable` status and be included in the release.
+
+In some cases, the questions below should also have answers in `kep.yaml` . This
+is to enable automation to verify the presence of the review, and to reduce review
+burden and latency.
+
+The KEP must have a approver from the
+[ `prod-readiness-approvers` ](http://git.k8s.io/enhancements/OWNERS_ALIASES)
+team. Please reach out on the
+[#prod-readiness](https://kubernetes.slack.com/archives/CPNHUMN74) channel if
+you need any help or guidance.
+-->
+
+### Feature Enablement and Rollback
+
+<!--
+This section must be completed when targeting alpha to a release.
+-->
+
+###### How can this feature be enabled / disabled in a live cluster?
+
+<!--
+Pick one of these and delete the rest.
+
+Documentation is available on [feature gate lifecycle] and expectations, as
+well as the [existing list] of feature gates.
+
+[feature gate lifecycle]: https://git.k8s.io/community/contributors/devel/sig-architecture/feature-gates.md
+[existing list]: https://kubernetes.io/docs/reference/command-line-tools-reference/feature-gates/
+-->
+
+* [x] Feature gate (also fill in values in `kep.yaml`)
+  + Feature gate name: PodObservedGenerationTracking
+  + Components depending on the feature gate: Kubelet
+
+###### Does enabling the feature change any default behavior?
+
+<!--
+Any change of default behavior may be surprising to users or break existing
+automations, so be extremely careful here.
+-->
+
+The pod's `metadata.generation` field is currently unset and `observedGeneration`
```

Comment:
I *think* it is only unset by default, but verify that

```suggestion
The pod's `metadata.generation` field is currently unset by default, and `observedGeneration`
```

Reply:
yeah you are correct, that's what I meant but wasn't specific. thanks!